helper/src/main/java/me/lucko/helper/utils/XMaterial.java

```diff
@@ -0,0 +1,1686 @@
+                    oldMat = Material.getMaterial(legacyName);
+                    if (oldMat != null) return oldMat;
+                } else return null;
+                // According to the suggestion format list, all the other names continuing
+                // from here are considered as a "suggestion" if there's no slash anymore.
+            }
+            oldMat = Material.getMaterial(legacyName);
+            if (oldMat != null) return oldMat;
+        }
+        return null;
+    }
+
+    /**
+     * Checks if an item is similar to the material and its data (if in older versions.)
+     *
+     * @param item item to check.
+     * @return true if the material is the same as the item's material (and data if in older versions.)
+     */
+    @SuppressWarnings("deprecation")
+    public boolean isSimilar(@Nonnull ItemStack item) {
+        Objects.requireNonNull(item, "ItemStack cannot be null");
+        Objects.requireNonNull(item.getType(), "ItemStack's material cannot be null");
+        return (isNewVersion() || this.isDamageable()) ? item.getType() == this.parseMaterial() : item.getType() == this.parseMaterial() && item.getDurability() == this.data;
+    }
+
+    /**
+     * Get the suggested material names that can be used instead of this material.
+     *
+     * @return a list of suggested material names.
+     */
+    @Nonnull
+    public String[] getSuggestions() {
+        if (this.legacy.length == 0 || !this.legacy[0].contains(".")) return new String[0];
+        return Arrays.stream(this.legacy).filter(mat -> !mat.contains(".")).toArray(String[]::new);
+    }
+
+    /**
+     * Checks if this material is supported in the current version.<br>
+     * It'll check both the newest matetrial name and for legacy names.
+     *
+     * @return true if the material exists in {@link Material} list.
+     */
+    public boolean isSupported() {
+        return Arrays.stream(Material.values()).anyMatch(mat -> mat.name().equals(this.name()) || this.matchAnyLegacy(mat.name()));
+    }
+
+    /**
+     * Gets the added version if the material is newly added after the 1.13 Aquatic Update and higher.
+     *
+     * @return the version which the material was added in. {@link MinecraftVersion#UNKNOWN} if not new.
+     * @see #isNew()
+     */
+    @Nonnull
+    public MinecraftVersion getVersionIfNew() {
+        return this.isNew() ? valueOfVersion(this.legacy[0]) : MinecraftVersion.UNKNOWN;
+    }
+
+    /**
+     * Checks if the material is newly added after the 1.13 Aquatic Update.
+     *
+     * @return true if it was newly added.
+     */
+    public boolean isNew() {
+        return this.legacy.length != 0 && this.legacy[0].contains(".");
+    }
+
+    /**
+     * Gets the suggested material instead of returning null for unsupported versions.<br>
+     * This is somehow similar to what ProtcolSupport and ViaVersion are doing to new materials.<br>
+     * Don't use this if you want to parse to a {@link Material}
```

Comment:
Additional paragraphs in comments should include an empty line between previous paragraphs.